### ISSUE TITLE
Override project image with branch-specific image when a Dockerfile is present

### ISSUE THREAD

**Issue by lmcmicu** [ontodev/droid#41]
*Currently*, at startup, the server will look in `config.edn` for the image to use with a given project and will either try to build it or pull it. If there happens to also be a Dockerfile on the master branch (or on some other branch if there is no master), then DROID will build, otherwise it will pull.

*We need to change this* to the following instead: At startup, DROID will look in `config.edn` and _pull_ all of the per-project images. If, in addition, any of the branches in the project contain a Dockerfile, then we should build an image to be used by that branch and that branch alone based on the Dockerfile.

In other words, the per-project images are to be thought of as default images. A Dockerfile on a branch will override this default image for that particular branch.

Note that in addition to performing this check at startup, we will also need to do it whenever a new branch is created or checked out.

**Comment by jamesaoverton** [comment]
I think that this should only apply to a `Dockerfile` at the top-level of the repository. The current code uses `file-seq` (which walks the file system tree) and is finding `Dockerfile`s that don't want to use.

https://github.com/ontodev/droid/blob/master/src/droid/command.clj#L36

**Comment by lmcmicu** [comment]
What you suggest seems like it will be easier. It (might, though I'll have to think about it) require some coordination between `config.edn` and the `Dockerfile` (in particular I'm thinking about the image name), but even it that's true it doesn't seem to me like a big deal.

I'm happy to do it this way if it satisfies the requirements of the use cases you're envisioning.
